tests/e2e/utils/helpers.js

```diff
@@ -32,3 +32,51 @@ export const checkPageExists = async ( slug ) => {
 		return Promise.reject();
 	}
 };
+
+/**
+ * Retrieves the product price from the current product page.
+ *
+ * This function assumes that the Puppeteer page object is already navigated to a product page.
+ * It extracts the textual content of the element with the class '.woocommerce-Price-amount.amount',
+ * which is expected to contain the product's price. The function then removes any non-numeric characters
+ * from this text, typically to exclude the currency symbol, and returns just the numeric price.
+ *
+ * @return {Promise<string>} A promise that resolves to the product price as a string. If the price element
+ *                            is not found on the page, the promise is rejected with an error message.
+ * @throws {Promise<Error>} If the price element is not found on the page, the function rejects the promise
+ *                          with an error message indicating that the price element was not found.
+ */
+export const getProductPriceFromProductPage = async () => {
+	await page.waitForSelector( '.woocommerce-Price-amount.amount', {
+		timeout: 5000,
+	} );
+	const price = await page.evaluate( () => {
+		let priceElement = document.querySelector(
+			'ins .woocommerce-Price-amount.amount'
+		);
+		if ( ! priceElement ) {
+			// If no discounted price is found, look for the regular price
+			priceElement = document.querySelector(
+				'.woocommerce-Price-amount.amount'
+			);
+		}
+
+		if ( priceElement ) {
+			let priceText = priceElement.textContent || '';
+
+			// Remove non-numeric characters (excluding the decimal point)
+			priceText = priceText.replace( /[^0-9.]/g, '' );
```

Comment:
Do you think worths adding the thousand separator (`,`) to this regex?

Reply:
Not required. The only concern would be if the decimal separator is a comma, but for our current test cases, this approach should suffice.